utils/src/test/scala/it/pagopa/interop/commons/utils/TypeConversionsSpec.scala

```diff
@@ -89,6 +94,44 @@ class TypeConversionsSpec extends AnyWordSpecLike with Matchers with ScalaFuture
             .of(2021, 1, 1, 10, 10, 10, 0, ZoneOffset.UTC)
         )
     }
+  }
+
+  "A future" should {
+    "have limited parallelism if using the correct api" in {
+      val tp: ExecutorService           = Executors.newFixedThreadPool(21)
+      implicit val ec: ExecutionContext = ExecutionContext.fromExecutor(tp)
+      val oneToTwenty: List[Int]        = (1 to 20).toList
+      def blockingFuture: Future[Unit]  = Future(Thread.sleep(1000))
+
+      def allTheLimitedFutures(): Future[List[Unit]] =
+        Future.traverseWithLatch[Int, Unit](10)(oneToTwenty)(_ => blockingFuture)
+
+      def allTheFutures(): Future[List[Unit]] =
+        Future.traverse(oneToTwenty)(_ => blockingFuture)
+
+      assert(!allTheLimitedFutures().isReadyWithin(1000.millis), "The limited future completed too early")
+      assert(allTheLimitedFutures().isReadyWithin(2100.millis), "The limited future completed too slowly")
+      assert(allTheFutures().isReadyWithin(1100.millis))
```

Comment:
flakiness is around the corner, but let' try like this for now 👍

Reply:
Well, I specifically gave him 21 threads, 1 to submit the traverse and the other to be blocked by all the `Thread.sleep`s. I hope they'll be enough.